client/lib/screen-title/index.js

```diff
@@ -13,16 +13,19 @@ import { setTitle as legacySetTitle } from './actions';
 export function subscribeToStore( store ) {
 	let title = store.getState().page.title;
 	let count = store.getState().page.unreadCount;
+	let siteId = store.getState().ui.selectedSiteId;
 
 	store.subscribe( () => {
 		let nextTitle = store.getState().page.title;
 		let nextCount = store.getState().page.unreadCount;
+		let nextSiteId = store.getState().ui.selectedSiteId;
 
-		if ( nextTitle !== title || nextCount !== count ) {
+		if ( nextTitle !== title || nextCount !== count || nextSiteId !== siteId ) {
```

Comment:
I worry about this addition a bit because, unlike before where a change could only be introduced to the `ScreenTitle` store if a section explicitly opted in to assigning a title via the Redux actions, this behavior would cause the title to be set on any page transition which incurs a change in site. 

I haven't spent enough time to think of a viable alternative, though perhaps introducing a function for updating the site option without it affecting the count or title would be an option? Similar to the `setCount` function in `lib/screen-title/actions.js`.